app/Console/Commands/SendTestEmail.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Mail;
+
+class SendTestEmail extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'monica:test-email
+                            {--email= : The email address to send to}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Send a test email to check that delivery is working';
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        // retrieve the email from the option
+        $email = $this->option('email');
+
+        // if no email was passed to the option, prompt the user to enter the email
+        if (! $email) {
+            $email = $this->ask('What email address should I send the test email to?');
+        }
+
+        // only perform basic validation, the mail server can handle the rest
+        if (strpos($email, '@') === false) {
+            $this->error('Invalid email address - missing "@" symbol!');
+
+            return -1;
+        }
```

Comment:
You could use `filter_var` - see https://www.php.net/manual/en/filter.examples.validation.php
```suggestion
        if (filter_var($email, FILTER_VALIDATE_EMAIL) === false) {
            $this->error("Invalid email address: '$email'.");

            return -1;
        }
```

Reply:
I've done this, but it fails Psalm for `PossiblyInvalidCast` `non-empty-array<array-key, mixed> cannot be cast to string (see https://psalm.dev/190)` in the string interpolation.

If I insert a deliberate cast for `$email = (string) $this->option('email');` instead, it fails on that line with `array<array-key, mixed> cannot be cast to string (see https://psalm.dev/190)`.

Any guidance? I've tried a few options and looked online but there was nothing obvious.